src/VirtoCommerce.CatalogCsvImportModule.Data/Services/CsvCatalogExporter.cs

```diff
@@ -45,89 +45,151 @@ IBlobUrlResolver blobUrlResolver
         }
 
         public async Task DoExportAsync(Stream outStream, CsvExportInfo exportInfo, Action<ExportImportProgressInfo> progressCallback)
-        {
-            var prodgressInfo = new ExportImportProgressInfo
+        {
+
+            var progressInfo = new ExportImportProgressInfo
             {
-                Description = "loading products..."
-            };
-
+                Description = "counting products...",
+                TotalCount = await GetProductsCount(exportInfo)
+            };
+
+            // It seems we need to read all the products twice. 
+            progressInfo.Description = "collecting product properties...";
+            progressCallback(progressInfo);

```

Comment:
the description changed twice, but progressCallback sent only once. Remove one of the descriptions?